packages/turbo-workspaces/src/utils.ts

```diff
@@ -0,0 +1,119 @@
+import fs from "fs-extra";
+import path from "path";
+import glob from "fast-glob";
+import yaml from "js-yaml";
+import semver from "semver";
+import { PackageJson, Workspace } from "./types";
+import { ConvertError } from "./errors";
+
+// adapted from https://github.com/nodejs/corepack/blob/main/sources/specUtils.ts#L14
+const PACKAGE_MANAGER_REGEX = /^(?!_)(.+)@(.+)$/;
+
+function getPackageJson({
+  workspaceRoot,
+}: {
+  workspaceRoot: string;
+}): PackageJson {
+  const packageJsonPath = path.join(workspaceRoot, "package.json");
+  try {
+    return fs.readJsonSync(packageJsonPath, "utf8");
+  } catch (_) {
+    throw new ConvertError(`no "package.json" found at ${workspaceRoot}`);
+  }
+}
+
+function getWorkspacePackageManager({
+  workspaceRoot,
+}: {
+  workspaceRoot: string;
+}): string | undefined {
+  const { packageManager } = getPackageJson({ workspaceRoot });
+  if (packageManager) {
+    const match = packageManager.match(PACKAGE_MANAGER_REGEX);
+    if (match && match.length === 3 && semver.valid(match[2])) {
+      return match[1];
+    }
```

Comment:
6 is afraid of 7 because 7 8 9.

Can we do `const [major, minor, patch] = match;` or something to that effect?

Reply:
destructuring > what the hell index is this? 😜